### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 10

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#250] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152315388-2410b9e0-47d1-4ccc-bb6e-519626dfc8b2.png)

**Comment by Elia-Belli** [comment]
Definiamo f: X->Y e g: Y->Z:
Se g o f è iniettiva allora ogni elemento di X deve avere immagine distinta in Z
Quindi segue dall'ipotesi che #X = #Im(f) $\subseteq$ Y = #Im(g) $\subseteq$ Z

- Allora f è necessariamente iniettiva, altrimenti la sua immagine avrebbe cardinalità inferiore a X.

- Non è detto che il dominio di g abbia stessa cardinalità di Im(f), quindi g potrebbe associare elementi appartenenti a Y ma non all'immagine ad uno stesso elemento di Z.
Anche se g rimane "iniettiva" sugli elementi di Im(f), sufficiente affinchè g o f sia iniettiva, potrebbe non esserlo su altri.

Quindi possiamo dire che **f è necessariamente iniettiva** e che **g è necessariamente iniettiva solo sugli elementi di Im(f)**.
